pandas/core/internals/blocks.py

```diff
@@ -1796,13 +1797,35 @@ def _can_hold_element(self, element: Any) -> bool:
         return True
 
     def _slice(self, slicer):
-        """ return a slice of my values """
-        # slice the category
+        """
+        Return a slice of my values.
+
+        Parameters
+        ----------
+        slicer : slice, ndarray[int], or a tuple of these
+            Valid (non-reducing) indexer for self.values.
+
+        Returns
+        -------
+        np.ndarray or ExtensionArray
+        """
         # return same dims as we currently have
+        if not isinstance(slicer, tuple) and self.ndim == 2:
+            # reached via getitem_block via _slice_take_blocks_ax0
```

Comment:
Could also put this check in `getitem_block` or `_slice_take_blocks_ax0` to ensure that `_slice` always gets the correct  tuple of slices?

Reply:
Or, directly check the value? As here you convert it to a len-2 tuple, and then in a next step, you convert the tuple back to a slice?

Reply:
Hmm one way to avoid this PITA would be to have 2D EAs...